cmd/cli/plugin/package/test/lib/package_plugin.go

```diff
@@ -400,10 +400,7 @@ func (p *packagePlugin) CheckRepositoryAvailable(o *packagedatamodel.RepositoryO
 			RepositoryName: o.RepositoryName,
 			Namespace:      o.Namespace,
 		})
-		if result.Error != nil {
-			return false, result.Error
-		}
-		if result.Stdout != nil && strings.Contains(result.Stdout.String(), "Reconcile succeeded") {
+		if result.Error == nil && result.Stdout != nil && strings.Contains(result.Stdout.String(), "Reconcile succeeded") {
```

Comment:
Just to be clear, does this change mean we don't return an error on repository Get error as we want to retry and only return error on time out to success?

Reply:
Yeah.. Since we also want to retry in case of error (to let the test work on slow testbed), so no Get error will be returned except for timeout error.

Reply:
Hi @maralavi, I removed the retries here since we've already done a retry on packaging API availability [here](https://github.com/vmware-tanzu/tanzu-framework/blob/main/cmd/cli/plugin/package/test/package_plugin_integration_test.go#L491-L504). But this block doesn't guarantee that the packaging API is available after `kapp-controller` restarts. So I updated my fix to let this block ensure the `kapp-controller` finishes restart and the packaging API is available. Details can be seen at the description of this PR. Thanks

Reply:
I see. Thanks for the clarification :)